ocaml/xapi/xapi_vdi.ml

```diff
@@ -116,7 +129,7 @@ let check_operation_error ~__context ?(sr_records=[]) ?(pbd_records=[]) ?(vbd_re
          snapshot's VBDs to true, and this would block operations that require
          the VDI to be detached. *)
       let my_active_vbd_records =
-        my_active_vbd_records |> List.filter
+        my_active_vbd_records |> Safe.filter
           (fun vbd ->
              let vm = vbd.Db_actions.vBD_VM in
              Db.is_valid_ref __context vm && (Db.VM.get_is_a_snapshot ~__context ~self:vm = false)
```

Comment:
Is there any point keeping the `Db.is_valid_ref` given that you're using `Safe.filter`, or does it just slow things down with an extra database call?

Reply:
Yes, for defence in depth. I wanted to be as safe as possible, because failures here will cause serious issues. See CA-255128